### PR TITLE
Add Staging deployment CI workflow

### PR DESCRIPTION
**Pull request by parth-07**

### CODE REVIEW

File: .github/workflows/staging-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    shell: bash
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```
Comment (manorit2001) [inline .github/workflows/staging-deploy.yml:19]:
```suggestion
          fetch-depth: 0
          ref: staging
```

File: .github/workflows/staging-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+      - name: Add heroku staging remote
+        run: heroku git:remote --app "${HEROKU_STAGING_APP_NAME}"
+        env:
+          HEROKU_STAGING_APP_NAME: ${{ secrets.HEROKU_STAGING_APP_NAME }}
+
+      - name: Push to heroku master
+        run: |
+          git checkout "${GITHUB_REF##*/}"
+          git push -f heroku "${GITHUB_REF##*/}":master
```
Comment (manorit2001) [inline .github/workflows/staging-deploy.yml]:
```suggestion
          git push -f heroku HEAD:master
```

**Review by manorit2001 — APPROVED** [review]
please merge the requested changes, everything else is fine

File: .github/workflows/staging-deploy.yml (hunk 1 above)
Comment (parth-07) [reply .github/workflows/staging-deploy.yml:19]: Made the changes.

File: .github/workflows/staging-deploy.yml (hunk 2 above)
Comment (parth-07) [reply .github/workflows/staging-deploy.yml]: Made the changes.

**Pull request merged**